components/contribution-flow/StepDetailsCrypto.js

```diff
@@ -26,14 +27,58 @@ const messages = defineMessages({
   },
 });
 
-const StepDetailsCrypto = ({ onChange, data }) => {
+/*
+ * Calculates the approximate value for 1 unit of a given crypto currency in the collective currency.
+ * Uses the Cryptonator API (https://www.cryptonator.com/api/)
+ */
+const getCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+  if (!cryptoCurrency || !collectiveCurrency) {
+    return null;
+  }
+  let body;
+  try {
+    const response = await fetch(`https://api.cryptonator.com/api/ticker/${cryptoCurrency}-${collectiveCurrency}`);
+    body = await response.json();
+    if (!body.success || body.error !== '') {
+      throw new Error(`Cryptonator Error: ${body.error}`);
+    }
```

Comment:
I would take the `return` in the `try .. catch`, to make sure we don't crash if the response content ever changes

```suggestion
    const body = await response.json();
    if (!body.success || body.error !== '') {
      throw new Error(`Cryptonator Error: ${body.error}`);
    }
    return body.ticker.price;
```

Reply:
Makes sense. 👍🏽 Thanks... I've refactored it.